### ISSUE TITLE
Improve interactive button styling

### ISSUE THREAD

**Issue by reefdog** [PhilanthropyDataCommons/data-viewer#173]
[Buttons](https://pilot.philanthropydatacommons.org/storybook/?path=/docs/stories-button--docs) are currently very static: there is no hover state to indicate their interactivity and no alternate "active" state when being depressed. There is only a very clumsy `:focus` ring for minimal accessibility, but it is a thick garish blue outline that rarely fits with the aesthetic of the particular button style.

We should at least add hover styling and theme-specific focus styling, plus consider an active style.
